`Development`: Add export documentation (#7249)
ls1intum/Artemis: docs/user/exports.rst

.. list-table:: Export options
   :widths: 100
   :header-rows: 1

   * - Export/Archive option
   * - Archive course/exam
   * - Export programming exercise material
   * - Export programming exercise student repositories
   * - Export exercise results
   * - Export quiz questions
   * - Export exercise submissions
   * - Export user data

Archive course/exam
-------------------
Export all course/exam data, including all exercises and student submissions.
To archive a course or an exam the end date of the entity needs to be in the past.
You can archive a course by clicking |archive_course| on the course management overview page or an exam by clicking |archive_exam| on the exam checklist page. This will create a zip file containing all relevant data.
The creation is done asynchronously. You will receive a notification once the archive is ready to download. You can then download the archive by clicking |download_archive| on the course management overview page or the exam checklist page.

Export programming exercise material
------------------------------------

> Export the exercise material (template,solution,test, auxiliary repositories and problem statement and other relevant information such as points) of a programming exercise.

Author: We do not include the student scores. We include all information from the programming exercise object that isn't lazily loaded.

> Export the exercise material (template,solution,test, auxiliary repositories and problem statement and other relevant information such as points) of a programming exercise.

Reviewer: Maybe something like this=
```suggestion
Export the exercise material (template, solution, test, and auxiliary repositories as well as the problem statement and other general exercise information) of a programming exercise.
```

To export the material click the |download_exercise| button on the exercise details page.

Export programming exercise student repositories
------------------------------------------------
Export the student repositories (this can include the repositories for both graded and practice participations) of a programming exercise
To export the repositories click the |export| button and then the |download_repos| button on the |download_scores| page.

Export exercise results
-----------------------------------

> Export the results of a specific exercise. This is supported for all exercise types.

Reviewer 2: ```suggestion
Export the results of students for a specific exercise in a structured format (CSV, XLSX). This is supported for all exercise types.
```
```suggestion
Export the results of students for a specific exercise as a CSV file. This is supported for all exercise types.
```
Suggestions depending on which file types can actually be exported.

To export the results click the |export| button and then the |export_results| button on the |download_scores| page.

> Export quiz exercise

Reviewer 2: Similar to exporting the programming exercise material, could be reordered to be after that section.

--------------------
Exports the questions and the sample solution of a quiz in JSON format
You can export a a quiz exercise by clicking the |export_quiz| button on the exercises overview page.

> Export exercise submissions

Reviewer 2: Nitpick: reordering the sections so that this one is before/after the ‘Export programming exercise student repositories’ ones would make sense since those do the essentially same thing with some adaptation for different exercise types.

---------------------------
Export the submissions of all students that participated in a specific exercise. This is supported for text, modeling and file upload exercises.
Text submissions are exported as a zip file containing all submissions as text files.
Modeling submissions are exported as a zip file containing all submissions as json files.
File upload submissions are exported as a zip file containing all submitted files.
To export the submissions click the |export_scores| button and then the |export_submissions| button on the |download_scores| page.

> Export course/exam scores

Reviewer 2: nitpick: reordering to be after ‘Export exercise results‘ would make sense to me since it is similar functionality.

-------------------------
Export the scores of all students that participated in a specific course or exam. This is supported for all exercise types.
The scores are exported in CSV format.
To export the scores of a course click on the |scores_navigation_bar| tab in the course management navigation bar and then the |export_scores| button.
For exams you can export the scores by clicking the |download_scores| button on the exam checklist page and then |export_scores| button.

Export user data
----------------
Export all data Artemis stores about a specific user. This includes information such as name or email, exercise submissions, results, feedbacks the user received, messages they've sent.

> You can request a data export by clicking |privacy_statement| and |data_export|. Once the export has been created you will receive an email with a download link.

Reviewer 2: ```suggestion
You can request a data export by clicking |privacy_statement| in the page footer and |data_export|. Once the export has been created you will receive an email with a download link.
```
It might not be obvious to users where this link is.

-------------------
Export all course/exam data, including all exercises and student submissions.
To archive a course or an exam the end date of the entity needs to be in the past.

> You can archive a course by clicking |archive_course| on the course management overview page or an exam by clicking |archive_exam| on the exam checklist page. This will create a zip file containing all exercises of the exam or course and all student submissions. For a course all exams are exported as well. For each exercise the problem statement, a JSON file with exercise details such as points and the problem statement are exported.

Reviewer 2: ```suggestion
You can archive a course by clicking |archive_course| on the course management overview page or an exam by clicking |archive_exam| on the exam checklist page. This will create a zip file containing all exercises of the exam or course and all student submissions. For a course all exams are exported as well. For each exercise the a JSON file with exercise details such as points and the problem statement are exported.
```
2x problem statement
